circup/__init__.py

```diff
@@ -238,7 +244,7 @@ def __init__(
         resulting self.file value will be None, and the name will be the
         basename of the directory path.
 
-        :param str path: The path to the module on the connected
+        :param str path: The path or URL to the module on the connected
```

Comment:
It seems weird to need a URL here. Seems like the backend should handle that.

Reply:
The only place where Module objects get created is in find_modules() https://github.com/vladak/circup/blob/web_workflow/circup/__init__.py#L596

which is creating a list of the Module objects that include details specific to the instance found on the target device connected via whichever backend is in use, like version of the instance that is installed.

find_modules() in turn is used by the Click CLI functions for the circup commands `list`, `freeze`, and `update` all of which are relying on the version and other information that was gleaned from this specific instance of the module found on the Backend's device. 

The path or URL that gets passed in to the Module constructor gets stored as `self.path` which is then used by several places within __init__ and backends, so in many of those cases this is how the backend is currently handling it.
![image](https://github.com/adafruit/circup/assets/2406189/25e95afa-b72a-4877-9f9d-2358b753ef94)

If we have the Module not represent the specific instance found via the Backend and don't pass in and store it's path or URL I'm not sure the Module class will have much more reason to exist. I'm not particularly confident that my attempt at replacing it with something different would turn out better than what's here now, but I'd be willing to give it a try if you think it is best to move away from using that class.

My inclination after digging around in the existing code and thinking about this for a while is to keep the path / URL argument as it is now, but move `find_modules()` to the Backend's which then means the only place these Module objects are being instantiated is inside of a Backend function. Which I think makes sense given that it's pulling information specific to the instance on the device. Then Module could be moved to be a sub class of Backend and/or renamed to BackendModule to make it clear that represents the specific instances on the device which the Backend is interfacing with. What do you think of that approach @tannewt?

Reply:
The latest commit changes Module() constructor to accept a `backend` and the `module_name` instead of the full path to the module. Inside of Module it now uses the backend to get the full path when needed.